src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/Microsoft.NET.Sdk.Razor.SourceGenerators.Test.csproj

```diff
@@ -34,7 +34,16 @@
   <ItemGroup>
     <ProjectReference Include="..\..\Microsoft.NET.Sdk.Razor.SourceGenerators\Microsoft.NET.Sdk.Razor.SourceGenerators.csproj" />
     <ProjectReference Include="..\Microsoft.AspNetCore.Razor.Test.Common\Microsoft.AspNetCore.Razor.Test.Common.Compiler.csproj" />
-    <ProjectReference Include="..\Microsoft.AspNetCore.Razor.Test.MvcShim\Microsoft.AspNetCore.Razor.Test.MvcShim.Compiler.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
```

Comment:
I'm slightly concerned that this will make our tests dependent on the version of .NET the repo is built against.

Reply:
That's true. I don't need the full ASP.NET Core framework, only the assemblies, like `Microsoft.AspNetCore.Mvc`. Unfortunately, they are not distributed normally as NuGet packages - otherwise I could just grab those at some fixed version.

Reply:
It's a new variable to introduce to our tests that makes them unreliable. I would prefer us not do that. We should reach out the aspnet team and see if they have a better solution.

Reply:
> I'm slightly concerned that this will make our tests dependent on the version of .NET the repo is built against.

Our tests are already dependent on the .NET Core version we build against. I mean we build against say `net7.0` but can run against that or `net8.0`. What distinction are you making that I'm missing here?

Reply:
> Our tests are already dependent on the .NET Core version we build against.

With `FrameworkReference`, the tests end up referencing ASP.NET Core assemblies at the same version that's specified in our `global.json` (currently 8.0.100-preview.1.23115.2). And that ASP.NET Core runtime is used to render the CSHTML into HTML.

> It's a new variable to introduce to our tests that makes them unreliable. I would prefer us not do that.

So, would it be fine if we used ASP.NET Core DLLs from a specific version, not tied to our `global.json`, say [7.0.4](https://www.nuget.org/packages/Microsoft.AspNetCore.App.Ref/7.0.4)? I think we could try some MSBuild magic to do that.

Reply:
> What distinction are you making that I'm missing here?

To use the roslyn analogy, it would be like every compilation picking up the reference assemblies from the host process, rather than specifying that we're compiling against version "X" of the reference assemblies.